File: trunk/emsi/asp-net-app/WebEMSOF/component/db/Class_db_emsof_request_return_comments.pas
unit Class_db_emsof_request_return_comments;

interface

uses
  Class_db;

type
  TClass_db_emsof_request_return_comments = class(TClass_db)
  strict private
  published
    //
    constructor Create;
    //
    function Bind
      (
      target: system.object;
      master_id: string
      )
      : boolean;
    procedure Log
      (
      master_id: string;
      demoter: string;
      content: string
      );
    //
  end;

implementation

uses
  mysql.data.mysqlclient,
  system.web.ui.webcontrols;

constructor TClass_db_emsof_request_return_comments.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
end;

function TClass_db_emsof_request_return_comments.Bind
  (
  target: system.object;
  master_id: string
  )
  : boolean;
var
  dr: mysqldatareader;
begin
  Bind := FALSE;
  self.Open;
  dr := mysqlcommand.Create
    (
    'select timestamp'
    + ' , demoter'
    + ' , content'
    + ' from emsof_request_return_comment'
    + ' where emsof_request_master_id = "' + master_id + '"'
    + ' order by timestamp',
    connection
    )
    .ExecuteReader;
  if dr.HasRows then begin
    GridView(target).datasource := dr;
    GridView(target).DataBind;
    Bind := TRUE;
  end;
  self.Close;
end;

procedure TClass_db_emsof_request_return_comments.Log
  (
  master_id: string;
  demoter: string;
  content: string
  );
begin
  self.Open;
  mysqlcommand.Create
    (
    'insert emsof_request_return_comment'
    + ' set emsof_request_master_id = "' + master_id + '"'
    +   ' , demoter = "' + demoter + '"'
    +   ' , content = "' + content + '"',
    connection
    )
    .ExecuteNonquery;
  self.Close;
end;

end.

File: asp-net-app/WebEMSOF/component/db/Class_db_emsof_request_return_comments.cs
using MySql.Data.MySqlClient;
using System;

using System.Web.UI.WebControls;
using Class_db;

namespace Class_db_emsof_request_return_comments
{
    public class TClass_db_emsof_request_return_comments: TClass_db
    {
        //Constructor  Create()
        public TClass_db_emsof_request_return_comments() : base()
        {
            // TODO: Add any constructor code here

        }
        public bool Bind(object target, string master_id)
        {
            bool result;
            MySqlDataReader dr;
            result = false;
            Open();
            using var my_sql_command = new MySqlCommand("select timestamp" + " , demoter" + " , content" + " from emsof_request_return_comment" + " where emsof_request_master_id = \"" + master_id + "\"" + " order by timestamp", connection);
            dr = my_sql_command.ExecuteReader();
            if (dr.HasRows)
            {
                ((target) as GridView).DataSource = dr;
                ((target) as GridView).DataBind();
                result = true;
            }
            Close();
            return result;
        }

        public void Log(string master_id, string demoter, string content)
        {
            Open();
            using var my_sql_command = new MySqlCommand("insert emsof_request_return_comment" + " set emsof_request_master_id = \"" + master_id + "\"" + " , demoter = \"" + demoter + "\"" + " , content = \"" + content + "\"", connection);
            my_sql_command.ExecuteNonQuery();
            Close();
        }

    } // end TClass_db_emsof_request_return_comments

}
